### PR TITLE
Add documentation for upcoming Terraform Cloud secret engine

### PR DESCRIPTION
**Pull request by catsby**
This is a work-in-progress PR to add documentation for the upcoming Terraform Cloud Secret engine. The documentation reflects this companion PR: https://github.com/hashicorp/vault-plugin-secrets-terraform/pull/3

Prerequisites:

- [x] Merge https://github.com/hashicorp/vault-plugin-secrets-terraform/pull/3 
- [x] Merge https://github.com/hashicorp/vault/pull/10931 to bundle hashicorp/vault-plugin-secrets-terraform into Vault

### CODE REVIEW

File: website/content/api-docs/secret/terraform.mdx [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+  use. This token must have the needed permissions to manage all Organization,
+  Team, and User tokens desired for this mount.
+
+
+### Sample Payload
+
+```json
+{
+  "address": "https://app.terraform.io",
+  "token": "glhf..."
```
Comment (Valarissa) [inline website/content/api-docs/secret/terraform.mdx:44]: lulz

File: website/content/docs/secrets/terraform.mdx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+token at any given time**. Generating a new Organization or Team API token by
+reading the credentials in Vault or otherwise generating them on
+[app.terraform.io](https://app.terraform.io) will effectively revoke **any**
+existing API token for that Organization or Team. 
+
+Due to this behavior, Organization and Team API tokens created by Vault will be
+stored and returned on future requests, until the credentials get rotated. This
+is to prevent unintentional revocation of tokens that are currently in-use.
+
+Below is an example of creating a Vault role to manage manage an Organization
```
Comment (Valarissa) [inline website/content/docs/secrets/terraform.mdx:105]: double manage. For all your management needs.

File: website/content/api-docs/secret/terraform.mdx (hunk 1 above)
Comment (catsby) [reply website/content/api-docs/secret/terraform.mdx:44]: 😄

File: website/content/docs/secrets/terraform.mdx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+---             -----
+organization    hashicorp-vault-testing
+role            testing
+token           <example token>
+token_id        at-fqvtdTQ5kQWcjUfG
+```
+
+### User roles 
+
+Traditionally Vault secret engines creates dynamic users and dynamic credentials
```
Comment (Valarissa) [inline website/content/docs/secrets/terraform.mdx:131]: Should this be `Traditionally, Vault secret engines create dynamic users and dynamic credentials`?

Comment (Valarissa) [reply website/content/docs/secrets/terraform.mdx:131]: Comma can be ignored, this is more the `creates` -> `create`

File: website/content/docs/secrets/terraform.mdx [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+along with them. At the time of writing, the Terraform Cloud API does not allow
+for creating dynamic users. Instead, the Terraform Cloud secret engine creates
+dynamice User API tokens by configuring a Vault role to manage an existing
```
Comment (Valarissa) [inline website/content/docs/secrets/terraform.mdx:134]: `dynamice` -> `dynamic`

File: website/content/docs/secrets/terraform.mdx [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+Terraform Cloud user. The lifecycle of these tokens is managed by Vault and
+will auto expire according to the configured TTL and Max TTL of the Vault
```
Comment (Valarissa) [inline website/content/docs/secrets/terraform.mdx:136]: Should `Max` be capitalized? Feel free to ignore this one.

**Review by Valarissa — COMMENTED** [review]
Made a couple of observations. Feel free to change/reply/ignore.

File: website/content/docs/secrets/terraform.mdx (hunk 4 above)
Comment (catsby) [reply website/content/docs/secrets/terraform.mdx]: Fixed in 475b193a1

File: website/content/docs/secrets/terraform.mdx (hunk 3 above)
Comment (catsby) [reply website/content/docs/secrets/terraform.mdx]: Indeed, thanks! Updated in 475b193a1

File: website/content/docs/secrets/terraform.mdx (hunk 2 above)
Comment (catsby) [reply website/content/docs/secrets/terraform.mdx]: lol fixed in 475b193a1 😆

File: website/content/docs/secrets/terraform.mdx (hunk 5 above)
Comment (catsby) [reply website/content/docs/secrets/terraform.mdx]: Probably not, good catch. Fixed in e6135a9c7

**Review by Valarissa — APPROVED** [review]
LGTM~!

File: website/content/docs/secrets/terraform.mdx [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+
+## Quick Start
+
+Most secrets engines must be configured in advance before they can perform their
+functions. These steps are usually completed by an operator or configuration
+management tool.
+
+1.  Enable the Terraform Cloud secrets engine:
+
+    ```text
```
Comment (tomhjp) [inline website/content/docs/secrets/terraform.mdx:32]: Should this be a ```shell-session block?

File: website/content/docs/secrets/terraform.mdx [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+
+Key             Value
+---             -----
+organization    hashicorp-vault-testing
+role            testing
+token           <example token>
+token_id        at-fqvtdTQ5kQWcjUfG
+```
+
+### User roles
```
Comment (tomhjp) [inline website/content/docs/secrets/terraform.mdx:129]: There's quite a bit of duplication from the quick start section here. It's fine and thorough as-is, but if you wanted to shorten it I think that would also be OK.

File: website/content/api-docs/secret/terraform.mdx [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+| Method | Path                |
+| :----- | :------------------ |
+| `POST` | `/terraform/role/:name` |
+
+### Parameters
+
+- `name` `(string: <required>)` – Specifies the name for the Vault role. This is
+  part of the request URL.
+
+- `organization` `(string: "")` – Organization name to manage the single API
```
Comment (tomhjp) [inline website/content/api-docs/secret/terraform.mdx:120]: Do organization and team_id not conflict with each other as well?

File: website/content/api-docs/secret/terraform.mdx [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+$ curl \
+    --request DELETE \
+    --header "X-Vault-Token: ..." \
+    http://127.0.0.1:8200/v1/terraform/role/tfuser
+```
+
+## Rotate Role
+
+This endpoint rotates the credentials for a Terraform Cloud role that manages an
+Organization or Team.
```
Comment (tomhjp) [inline website/content/api-docs/secret/terraform.mdx:251]: What happens if you call it on a role that manages a user?

File: website/content/api-docs/secret/terraform.mdx [hunk 10]
```diff
@@ -0,0 +1,310 @@
...
+### Sample Request
+
+```shell-session
+$ curl \
+    --request POST \
+    --header "X-Vault-Token: ..." \
+    http://127.0.0.1:8200/v1/terraform/rotate-role/testing
+```
+
+## Generate Credential
```
Comment (tomhjp) [inline website/content/api-docs/secret/terraform.mdx:272]: nit: Would this be better titled as "Generate token"?

File: website/content/docs/secrets/terraform.mdx (hunk 6 above)
Comment (catsby) [reply website/content/docs/secrets/terraform.mdx:32]: Yeah probably. I updated this and others in 95d3d83a4

File: website/content/docs/secrets/terraform.mdx (hunk 7 above)
Comment (catsby) [reply website/content/docs/secrets/terraform.mdx:129]: I'll leave it as is. For docs I prefer verbosity

File: website/content/api-docs/secret/terraform.mdx (hunk 8 above)
Comment (catsby) [reply website/content/api-docs/secret/terraform.mdx:120]: They do not conflict. `team_id` takes precedence though

File: website/content/api-docs/secret/terraform.mdx (hunk 9 above)
Comment (catsby) [reply website/content/api-docs/secret/terraform.mdx]: It errors; I added a note about that in f3dbed26b

File: website/content/api-docs/secret/terraform.mdx (hunk 10 above)
Comment (catsby) [reply website/content/api-docs/secret/terraform.mdx:273]: I _believe_ `Credential` is consistent with other secret engines, so unless you strongly object I'd like to leave as is

**Review by tomhjp — APPROVED** [review]
LGTM!

**Pull request merged**